src/main.rs

```diff
@@ -0,0 +1,270 @@
+//! # Network Error Logging
+//!
+//! A Compute@Edge service which exposes a HTTP reporting endpoint for the
+//! W3C [Network Error Logging API](https://www.w3.org/TR/network-error-logging).
+use chrono::Utc;
+use fastly::error::anyhow;
+use fastly::http::{header, HeaderValue, Method, StatusCode};
+use fastly::log::Endpoint;
+use fastly::{downstream_client_ip_addr, uap_parse, Body, Error, Request, Response, ResponseExt};
+use serde::{Deserialize, Serialize};
+use std::fmt;
+use std::io::Write;
+use std::net::IpAddr;
+use std::str::FromStr;
+
+/// ReportBody models the body of a Network Error Log report which details the
```

Comment:
This is a small cosmetic nit that applies in other places, but I'll hang the comment here:

```suggestion
/// `ReportBody` models the body of a Network Error Log report.
///
/// A report body provides further details about a network error that occurred.
// ...
```

Could we slightly re-word comments like this so that the first sentence is a single-line? Rust's documentation uses the first sentence as the summary of an item in Rustdoc's output, so longer sentences like this don't format well when you open the `cargo doc` information up.

The API Documentation Conventions RFC is useful reference, [here](https://github.com/rust-lang/rfcs/blob/26197104b7bb9a5a35db243d639aee6e46d35d75/text/1574-more-api-documentation-conventions.md#summary-sentence) :blush: